include/bluetooth/mesh/cfg_srv.h

```diff
@@ -93,10 +103,13 @@ struct bt_mesh_cfg_srv {
 		 *              @ref BT_MESH_FEAT_PROXY,
 		 *              @ref BT_MESH_FEAT_FRIEND and
 		 *              @ref BT_MESH_FEAT_LOW_POWER.
+		 *
+		 * @deprecated Please use the @ref bt_mesh_hb_cb registration
+		 *             function.
 		 */
 		void (*func)(uint8_t hops, uint16_t feat);
 	} hb_sub;
-};
+} __deprecated;
```

Comment:
@jhedberg I'm unsure about the usage of deprecated here. I have marked this struct as deprecated, but it's still accepted in the composition data entry, and the init function still reads the feature flags, potentially overriding the new kconfig entries. Should I instead just flat out remove the structure right away?

Reply:
We should probably just remove it (and document it clearly as required by the API change process)

Reply:
Done. Also removed the parameter from the `BT_MESH_MODEL_CFG_SRV` macro. This breaks the pattern of having a model struct parameter to these `BT_MESH_MODEL_` macros.